#!/bin/bash

#####
#add_sample_name.sh
#
#This script is used to add Australian microbiome sampleID, plate ID and amplicon information to the definition lines of fasta formatted sequence files
#
#Input files are required to be in the format: 
#	sampleID_plateID.fasta
#
#The script will modify each fasta file by replace the > with >sampleID_plateID;<amplicon>_ in each sequence definition line
#
#Replace <amplicon> in the second last line with the appropriate amplicon (e.g., 16S, A16S, 18Sv4, 18Sv9, ITS1 or ITS2), for example:
#	`perl -pi -e 's/\>/\>sample='$VAR1';16S_/' $file;`
#####

#this initiates the loop for all files with extension .fasta in the list generated by ls -l

for file in `ls -1 *.fasta`

do

#tell me which file you're working on

echo working on $file

#cut the appropriate string from the name of the file.  -d means dlimited, "." delimited by ., -f 1 means the first bit that is delimited

VAR1=`echo $file | cut -d "." -f 1`

#tell me what you've cut and called VAR1

echo VAR1 is $VAR1

#use VAR1 to add the sample name to the sequence id.  Substitute > for >VAR1_ in all files.  The variable (VAR1) needs to be in quotation marks to be passed from bash to perl, the perl script needs to be embedded in the bash script with the ` ` marks

`perl -pi -e 's/\>/\>sample='$VAR1';<amplicon>_/' $file;`

done
